pkg/monitoring/metrics/virt-controller/vmstats_collector.go

```diff
@@ -587,3 +597,78 @@ func collectVMCreationTimestamp(vms []*k6tv1.VirtualMachine) []operatormetrics.C
 
 	return cr
 }
+
+func CollectVmsVnicInfo(vms []*k6tv1.VirtualMachine) []operatormetrics.CollectorResult {
+	var results []operatormetrics.CollectorResult
+
+	for _, vm := range vms {
+		if vm.Spec.Template == nil || vm.Spec.Template.Spec.Domain.Devices.Interfaces == nil {
+			continue
+		}
+
+		interfaces := vm.Spec.Template.Spec.Domain.Devices.Interfaces
+		networks := vm.Spec.Template.Spec.Networks
+
+		for _, iface := range interfaces {
+			bindingType, bindingName := getBinding(iface)
+			networkName, matchFound := getNetworkName(iface.Name, networks)
+
+			if !matchFound {
+				continue
+			}
+
+			results = append(results, operatormetrics.CollectorResult{
+				Metric: vmVnicInfo,
+				Labels: []string{
+					vm.Name,
+					vm.Namespace,
+					iface.Name,
+					bindingType,
+					networkName,
+					bindingName,
+				},
+				Value: 1.0,
+			})
+		}
+	}
+
+	return results
+}
+
+func getBinding(iface k6tv1.Interface) (bindingType, bindingName string) {
+	bindingType = none
+	bindingName = none
+
+	switch {
+	case iface.Masquerade != nil:
+		bindingType = "masquerade"
+	case iface.Bridge != nil:
+		bindingType = "bridge"
+	case iface.SRIOV != nil:
+		bindingType = "sriov"
+	case iface.Binding != nil:
+		bindingType = "binding"
+		bindingName = iface.Binding.Name
+	}
+
+	return bindingType, bindingName
+}
+
+func getNetworkName(ifaceName string, networks []k6tv1.Network) (networkName string, matchFound bool) {
+	networkName = none
+	matchFound = false
+
+	for _, net := range networks {
+		if net.Name == ifaceName {
+			matchFound = true
+			if net.Pod != nil {
+				networkName = "pod networking"
+			} else if net.Multus != nil {
+				networkName = net.Multus.NetworkName
+			}
+			break
+		}
+	}
```

Comment:
Using a variable like this is not nice.
```go
net := vmispec.LookupNetworkByName(networks, iface.Name)
if net != nil {
  switch {
  case net.Pod != nil: networkName = "pod network"
  case net.Multus != nil: networkName = net.Multus.NetworkName
  }
}
```

And consider calling it "pod network".

There are however some exceptions here, you better consider them in a follow up.
E.g. the pod network can be defined using Multus by setting it with a `default` flag.

Reply:
> it's called "pod networking" also in the UI so need to update both if changing.

Ok, so never-mind.



>> E.g. the pod network can be defined using Multus by setting it with a default flag.

> so you mean that the pod network can be Multus? so you mean maybe consider adding also the real pod network and not just showing "network: pod networking"

I mean that when `net.Multus` is set, it can be used to define the pod network as well. There is a special flag there, I think called `default`, that will do it.
See here: https://github.com/kubevirt/kubevirt/blob/3b6439796c27c94f994ea06c5530bc276e1651a2/staging/src/kubevirt.io/api/core/v1/schema.go#L1572

Reply:
so I think it still consider as `pod networking`  since it's determine by the pod. maybe we need to add also the pod network information if possible as new labels. not sure.

@sradco WDYT?

Reply:
@EdDev I'm not sure I understand, you mean that if net.Multus is set with Default flag then it is pod networking and not net.Multus? or that if net.Multus is set it can be overwritten to pod network by this default flag?